src/main/java/org/minions/devfund/benjamin/Squarelotron.java

```diff
@@ -0,0 +1,198 @@
+     * @param col column number.
+     * @return false if the element does not belog to the ring, eoc true.
+     */
+    private boolean isRingElement(int ring, int row, int col) {
+        int min = ring - 1;
+        int max = size - ring;
+        boolean inRow = row == min || row == max;
+        boolean inCol = col == min || col == max;
+        boolean rowInRange = min <= row && row <= max;
+        boolean colInRange = min <= col && col <= max;
+        return inRow && colInRange || inCol && rowInRange;
+    }
+
+    /**
+     * Horizontal flip within a 2D array.
+     * @param matrix object to be flipped.
+     * @return 2D array.
+     */
+    private int[][] horizontalFlip(final int[][] matrix) {
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size / 2; j++) {
+                int temp = matrix[i][j];
+                matrix[i][j] = matrix[i][size - j - 1];
+                matrix[i][size - j - 1] = temp;
+            }
+        }
+        return matrix;
+    }
+
+    /**
+     * Vertical flip.
+     * @param matrix object to be flipped.
+     * @return 2D array.
+     */
+    private int[][] verticalFlip(final int[][] matrix) {
+        for (int j = 0; j < size; j++) {
+            for (int i = 0; i <= size / 2; i++) {
+                int temp = matrix[i][j];
+                matrix[i][j] = matrix[size - i - 1][j];
+                matrix[size - i - 1][j] = temp;
+            }
+        }
+        return matrix;
+    }
+
+    /**
+     * Vertical flip of an specific ring.
+     * @param matrix object to be modified.
+     * @param ring an specific ring.
+     * @return 2D Array.
+     */
+    private int[][] verticalFlip(final int[][] matrix, int ring) {
+        for (int j = 0; j < size; j++) {
+            for (int i = 0; i <= (size - 1) / 2; i++) {
+                if (isRingElement(ring, i, j)) {
+                    int temp = matrix[i][j];
+                    matrix[i][j] = matrix[size - i - 1][j];
+                    matrix[size - i - 1][j] = temp;
+                }
+            }
+        }
+        return matrix;
+    }
+
+    /**
+     * Transpone a 2D array.
+     * @return 2D array.
+     */
+    private int[][] getTransponedMatrix() {
+        int[][] copy = new int[size][size];
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                copy[i][j] = squarelotron[j][i];
+            }
+        }
+        return copy;
+    }
+
+    /**
+     * Traspone only elements in an specific ring.
+     * @param ring an specific ring.
+     * @return 2D Array.
+     */
+    private int[][] getTransponedMatrix(int ring) {
+        int[][] copy = new int[size][size];
+        for (int i = 0; i < size; i++) {
+            for (int j = 0; j < size; j++) {
+                if (isRingElement(ring, i, j)) {
+                    copy[i][j] = squarelotron[j][i];
+                } else {
+                    copy[i][j] = squarelotron[i][j];
+                }
```

Comment:
This may be refactored to
`copy[i][j] = isRingElement(ring, i, j) ? squarelotron[j][i] : squarelotron[i][j];`